kernel/proc.c

```diff
@@ -7,7 +7,10 @@
 #include "proc.h"
 #include "spinlock.h"
 
-struct ptable_type ptable; /* initialize process table */
+struct {
```

Comment:
I didn't realize ptable_type was used both in pro.c and sysproc.c 

Since we don't want to have duplication (in this case of the type definition), I suggest we avoid this. 

Maybe just have `extern struct ptable_type ptable` in sysproc.c instead of proc.h